<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Reconstructing mult-omics networks with coglasso</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Reconstructing mult-omics networks with
coglasso</h1>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This vignette illustrates the basic usage of the coglasso package to
reconstruct a multi-omics network. The package implements an R interface
to <em>collaborative graphical lasso</em> (<a href="#ref">Albanese,
Kohlen and Behrouzi, 2024</a>), a network estimation algorithm based on
<em>graphical lasso</em> (<a href="#ref">Friedman, Hastie and
Tibshirani, 2008</a>) and <em>collaborative regression</em> (<a href="#ref">Gross and Tibshirani, 2015</a>)</p>
<p>Let us first attach coglasso.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(coglasso)</span></code></pre></div>
<p>We then choose the multi-omics data set to use. The coglasso package
offers three alternative version of a transcriptomics and metabolomics
data set. We will use <code>multi_omics_sd_small</code>. For further
explanation about the available multi-omics data sets see
<code>help(multi_omics_sd)</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">colnames</span>(multi_omics_sd_small)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;Cirbp&quot;       &quot;Hspa5&quot;       &quot;P4ha1&quot;       &quot;Spred1&quot;      &quot;Creld2&quot;     </span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt;  [6] &quot;Pdia6&quot;       &quot;Hsp90b1&quot;     &quot;Hsph1&quot;       &quot;Manf&quot;        &quot;Pdia3&quot;      </span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt; [11] &quot;Dnajb11&quot;     &quot;Dnajc3&quot;      &quot;BC004004&quot;    &quot;Stip1&quot;       &quot;Phe&quot;        </span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; [16] &quot;Trp&quot;         &quot;Putrescine&quot;  &quot;PC aa C36:3&quot; &quot;PC ae C32:2&quot;</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="fu">nrow</span>(multi_omics_sd_small)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt; [1] 30</span></span></code></pre></div>
<p>This smaller version of <code>multi_omics_sd</code> has 19 variables,
14 genes and 5 metabolites, and 30 samples. We can directly proceed with
network reconstruction.</p>
</div>
<div id="multi-omics-network-reconstruction" class="section level2">
<h2>Multi-omics network reconstruction</h2>
<p>Our objective is to reconstruct a network from this data set using
<em>collaborative graphical lasso</em>. To do so with the coglasso
package, we mainly call a function: <code>bs()</code>. This function
first estimates a network for every combination of hyperparameters we
want to explore, then it selects the best combination according to the
chosen model selection method.</p>
<p>The usual application of <code>bs()</code> requires to give an input
data set to the argument <code>data</code>, the number of variables of
the various omics layers <code>p</code>, and the hyperparameter
settings. Collaborative graphical lasso has three hyperparameters: <span class="math inline">\(λ_w\)</span>, penalizing “within” same-type
interactions, <span class="math inline">\(λ_b\)</span> penalizing
“between” different-type interactions, and <span class="math inline">\(c\)</span>, the weight of the collaborative term.
In this vignette we choose to explore 15 possible penalty values for
both “within” and “between” penalties, and three possible collaboration
values. We do so by setting both <code>nlambda_w</code> and
<code>nlambda_b</code> to 15, and by setting <code>nc</code> to 3. We
also decide to focus our search to the sparse side of possible “within”
networks. We achieve this by setting to a fixed value the ratio between
the smallest (least penalizing) and the largest (most penalizing) <span class="math inline">\(λ_w\)</span> explored. While the default value of
these parameter is 0.1 for both <span class="math inline">\(λ_w\)</span>
and <span class="math inline">\(λ_b\)</span>, we decide to set the
minimum ratio to 0.6 for <span class="math inline">\(λ_w\)</span>.</p>
<p>It is also possible to set options for the model selection procedure,
if one does not want to enjoy the comfort of the default behaviour. To
select the best combination of hyperparameters, we will set
<code>method</code> to “xestars” (which is the default behaviour). This
implements <em>eXtended Efficient StARS</em>, a significantly faster
version of <em>eXtended StARS</em> (<em>XStARS</em>, <a href="#ref">Albanese, Kohlen and Behrouzi, 2024</a>). <em>XEStARS</em>
and <em>XStARS</em> are a coglasso-adapted version of <em>StARS</em>,
the stability selection method developed by Liu, Roeder and Wasserman
(<a href="#ref">2010</a>). The suggested application of
<code>bs()</code> uses the default options of the function. For further
explanation on other selection methods available, and on other arguments
of <code>bs()</code> and how to use them, please see
<code>help(bs)</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>sel_cg <span class="ot">&lt;-</span> <span class="fu">bs</span>(</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  multi_omics_sd_small,</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="at">p =</span> <span class="fu">c</span>(<span class="dv">14</span>, <span class="dv">5</span>),</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="at">nlambda_w =</span> <span class="dv">15</span>,</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">nlambda_b =</span> <span class="dv">15</span>,</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">nc =</span> <span class="dv">3</span>,</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="at">lambda_w_min_ratio =</span> <span class="fl">0.6</span>,</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>)</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co"># To see information on the network estimation and selection:</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="fu">print</span>(sel_cg)</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; Selected network estimated with collaborative graphical lasso</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="co">#&gt; The call was:</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="co">#&gt; bs(data = multi_omics_sd_small, p = c(14, 5), nlambda_w = 15, </span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="co">#&gt;     nlambda_b = 15, nc = 3, lambda_w_min_ratio = 0.6, verbose = FALSE)</span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="co">#&gt; The model selection method was:</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a><span class="co">#&gt; xestars</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="co">#&gt; The density of the selected network is:</span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="co">#&gt; 0.06432749</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="co">#&gt; Networks are made of 2 omics layers, for a total of 19 nodes</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="co">#&gt; For each layer they have: 14 and 5 nodes, respectively</span></span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a><span class="co">#&gt; The selected value for lambda within is:</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a><span class="co">#&gt; 0.8276</span></span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a><span class="co">#&gt; The selected value for lambda between is:</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a><span class="co">#&gt; 0.5691</span></span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a><span class="co">#&gt; The selected value for c is:</span></span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a><span class="co">#&gt; 0.01</span></span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a><span class="co">#&gt; The total number of hyperparameter combinations explored was:</span></span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a><span class="co">#&gt; 675</span></span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a><span class="co">#&gt; The values explored for lambda within were:</span></span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a><span class="co">#&gt; 0.9137, 0.8933, 0.8722, 0.8503, 0.8276, 0.804, 0.7795, 0.7542, 0.7279, 0.7006, 0.6723, 0.643, 0.6125, 0.581, 0.5482</span></span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a><span class="co">#&gt; The values explored for lambda between were:</span></span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a><span class="co">#&gt; 0.5922, 0.5816, 0.5691, 0.5544, 0.5371, 0.5166, 0.4925, 0.4641, 0.4306, 0.3912, 0.3447, 0.2898, 0.2252, 0.149, 0.0592</span></span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a><span class="co">#&gt; The values explored for c were:</span></span>
<span id="cb3-41"><a href="#cb3-41" tabindex="-1"></a><span class="co">#&gt; 0.01, 1, 100</span></span>
<span id="cb3-42"><a href="#cb3-42" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-43"><a href="#cb3-43" tabindex="-1"></a><span class="co">#&gt; Plot the selected network with:</span></span>
<span id="cb3-44"><a href="#cb3-44" tabindex="-1"></a><span class="co">#&gt; plot(sel_cg)</span></span></code></pre></div>
<p>With this we have fast selected the combination of hyperparameters
yielding the most stable, yet sparse coglasso network. We can directly
plot the selected network with:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">plot</span>(sel_cg)</span></code></pre></div>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAB3VBMVEUAIGAAIGoAIG0AIHQAIHoAIH0AIH4AIJAAIJkAOG8AO2kAO4YAPn4AQmAAQrMAQ5EARosATHQATJkATLMAU5EAWs0AYGAAYM0AZ5kAdJkAdM0AfX0AfecAmJkAmM0AmOcAmP8AsrMAsv8AzM0AzOcAzP8XIGAXIH0hhucmIGAmIG0mU3ooIGAqIGAxIGAyIGAyIH0yYXk6IGA6IH06IJk6PmA6RYs6TH06TLM6VG06dJk6dM06jf86mbM6medEIGBEgKlYIGBYdH1YdLNflJFflKlmIGBmIH1mIJlmdJlmdM1mmbNmmedmpbNmvM1mvOdmvP95U2B5lIZ5qamAAACAADqAAGaAOpCAZraPAACQTGCQTJmQdLOQvLOQvP+QwbOQ3s2Q3v+Ram2RqamXAACXADqXVoOXZmaXkNujqampgHqplIapqZGpqZ2pqamtAACtZmatbZmtkNuttratttuttv+2dGC2dH22vJm2/822//+8mX3COgDCOjrCtpDCtv/C27bC2//WZgDW25DW/9vW///bmXPbmXTbmX3bmbPb/83b/+fb///rkDrrkJDr/7br////g1b/mZn/tmb/vJn/vLP/25D/3pn/3rP/3uf//7b//83//9v//+f///9rzMNPAAAACXBIWXMAAA7DAAAOwwHHb6hkAAALu0lEQVR4nO2diX8bRxmGh2CDuVvMURxakoLLfSlAwlHFBCIoR6oQJSmGlrgYCHFZAyUquCklhYAKKaR4CYsV79/KfHPsJcmvZEu7cvQ+v1iyd1e29snMtzPffrtSMdkTVfUbmHYoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgQHmCug1lWOpdFdWa+p/7YXd18Xb2uWLKbEGdhS39oOQx9xaEmnKCtEcjxj9XTemC4rC447ZhJS0o7rj1nRkVFLfn13LvQC1rlKIgL6ijmt3Gk6sSjDqm8SwfV8eU0kusuZkXFKrHG9pFOL8WnVjTP6njSgRpO22zwcwL0i1IjlliR/e3pAU1JTBLP5t5QToGGUH6S7sK59Mg3W3UYwoKdfBxgkK9wAky4yDbqGZckBkHOUEdHXfka8MOFHdXzSCy7cS0Z0yQH0nXzbfzT9fU/BW96Ct6iLhheptZZTaTo5l/rpqpmIttnJgCEwOYDkFf2MIbVcQUCAozo+jpYwoExUHVb2AvKAhAQQAKAlAQgIIAJQjaXU2O4zb/o3J/tD3qUf7ejZbwVN+VZlV2wc7m9ZF+e5FSWpDPsmpVzaKhdj0YedK1ffFmvN26fKfPqlbR0OEQ9IRtJOEHGj7B6ld1G83Re5gIil+RB8O9G4mDVuvfmnwbOhjlCHrSnsJ54UqPID2L36eg11peS6aRHFZBzVAm5tGpv4ugo6aP2bS0BKUjI89UnaBft1rXt1ut5zdbrUt/Xn9+89Kte7p7/VwEtS7dMrHqKbG3s/m7GxKyZMHVUf9WSYK6DUnRNyWHcXz+Ud2Crvi09H5b0N31y3furl/Xz9f1v22r5KVW6/X137yuW9A/b0nH2774p00RePGmfsm9l+RVo0akkgRJXqz77S0dcWyQdlnXE/sUJIFYx+hUkN3z7ZZDjnDbNoj79T+7adeO2oTKEqSbUFiPvaBrSVo6qq2M/vu2XXzuEXT5jpFwd10rei0vSLeyvoc9RFmCdBM6uWZOW4QLW0F4ZDKCLt1yzXFbRyfd5+KcILtgREoTFNWW7Hkd3dmCzngE7WxelYNZIujejcu/jLev3n1ZDwKu7mzqjnb3t6kgvfaOXjvi3yppJK2P8m2bd16UI9knaxKHnq4NKvfYEzOStuPoV1qtX6xf+osdWmtDOz+UNbqHyfodfXS7/K9NWXXxj+s6UMuC/uPvPahmLhZkJmDjnIlNYFZHQYCKZvMZQ2PcqUmkBaoT5HZnyv1QEKKqhFliaHx7NZm8W5WCzMOvVnPH+iiXLpK529DZtPtUUBDLTN+n1MJcylHP3nzNzJC/cfxUlpP2hqygpHBRqWNpvuh/Wlr3W8NlQyaV2K5WkH50gjquK+UECdFXe14qzWzxHz/2eVrT5Lqn7etlBjNSx0RUd1bDNiH9mAqSejx17MNSkOfDUnSqJ19tGltUS7qe65U2qaufd3+0lfg+OBULip2gqCazNf2l1COmpHPNe+upO3f7LvHJkC8j1i3oP2uxr8UaA5ULCmxg1m5MAV7kg3TbFwWbsqqUbsOdA3l1kCDz3D78gpI+Frr4LMerYMP6MdWdZvHuaj33qqhWT78911jYku0/aAW1k662u3r4u5g/jgVeUDh/NpAWsOGrO+3idn5Xw6RFhab77aZlxNJF21ZQn9C1T6oXFF9zJp47bcs7N9Lqzrh3V1NBtjv5kVOtabuo7WIvjK2mr8pz82acKOk0OWDpvubLPJPqzidscMphzo9YrKBFX0ZshpRWUGd8JWsVC8oMqDNLs/hBj63C1wY+7UOWvcYstEc5qbqe9y0orOtvnhnPm5wGQbr1BLmlGZJBjxMU1dQXTZhp160Mc91dWNffdxuuAtv0umFnKIgSBe0OmpY6Pe5Su5yg/KDHBZuOPVi5AbM8LUnl9Zrroif3lekeRKktqO+0NLANyfz/xwU/hUFPZshcGqUL6j8tDZLYmxOUGfSYc46PqIXjSpkLy6JacoHZRClfUN9padD2MSMnKD2m6+6jj2kLy3rMo+TCst9nLjCbKBUJKk5Lu6cfb/QJQflBj1F6XJTq5mQvD6pNvL+VLqj/tDRUp7Z2V2WukXtBftCTFVSP0gvMJkq5ggZOS8OFC7K6WRwFtbODnigJ0rYsxF+iOFHK72Kxu3DMHeafNdNSqWgwu1sQJFWNsR/0RO4wb5IZ5lx/CZeUVSLIj3mj2oo0nVCGMM3ANKyezGlm0KNZif2FZVHtZO88ZAJUIsiPebUgU+shU1PdhNpLMLOcri8hPhvKHknboJtOS79paz2aclx7D06939eC+pI9eRiMIEjPycoZUk+FoNTQ8C2oLKoWVDQ01NZlMiWChq1kmF1B6WmyobYuj8oFZQ3hKD3LgmyIpqAeMtnofHJ6z21Lo3pBxXz9XhJmXFAQoyY084KQIQoChmZTUGEoREE9BMWngR4oyD0PEjGjgooTsQGT+r1vBpvdapx5kOkQZHJpKyabKrvfz5AkgB6c63sz2KygkSqHh2E6BEke/0Kc7mgfQyLvzFzxZrC5snNhlMrhoZgKQVqHFZScxgmKe979jn7ozBVvBqt6DMV9K4f3z5QK6tlzcx8vJyi9GazZ6P32LozSTQdVDh+A8gSZONNM4kyO80q9WQT9zW2xoDf60Dv0w+rii2Z7m/B3gtKbwSr16PLD8+Zu1FK2Oahy+CCUfyfOPqcj2kvBX91pjnON73/pM+obcsJD7j+gHvicq5gWR2fmLgSaZ9XKxsdXgo1PnFVq+X1q3rSgMA1MxcrhA1G+oN7TxfaM6cLW7uq5RjN84LMPyWl4OQ+v1Jff9tPgvDpyVrwE59+axCB3zbRs9rAVlLmxabFy+EBMgyBXS9WMTlxp6Ief1NS7vKDP6y7WTE7IPjhXvBmsC1XuvrCe9hgHQhUJMoHC10BJJDn6JqXChT+cXnnu9MqG1CEaGmfenqmYDs8k4yB/M1izlZSM2LuDRN+T7cYapUsVpHx5huzCq0kNlFLvXD76RqWuzf1AdvRrekz4MbNhe/HFt6QV0/99zL683nszWH+Nz+LtUI35jH01LciUZsS+Bsp5U+Eb9MNDNe3hvXrxY7KnnXd/3Sj4bs3s96C5WDS5m8FW1MV0I7GBRGqg5HKUj+iDui0bOuEi8KfqcfGe0gPnqtHkbgZbWZAObccwP7pQa65A0UqsoI9mb75tGSBoojeDrWYcFD0jRVFJDZQfNeuxjO2ATV+b0B6uBU2QakbSsv9LSQ2U/qlut+lYLcWbb3vub0G9jFziQ0GAGRM0eg3UjAnak75zfyMomaqU8/la0yqo79xfBOnAbZeU9Pla0y6okEKLs8pK+WSSQyZIUVBCYe7vc4z2OigTl2ZdUH7u73OMW1HNZFtjCsrP/X0KLfPRWxQUZ+b+XlDdf/QWBRXm/i7H2KQgR3Hu73KMt1NBpXy+1rQKGjj399nWtZI+X2taBfVS1uU9BSgIcGgElXb9U4FDI6gqKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAjwf0BSf8P3HG7HAAAAAElFTkSuQmCC" /><!-- --></p>
</div>
<div id="ref" class="section level2">
<h2>References</h2>
<p>Albanese, A., Kohlen, W., &amp; Behrouzi, P. (2024). Collaborative
graphical lasso (arXiv:2403.18602). <em>arXiv</em> <a href="https://doi.org/10.48550/arXiv.2403.18602" class="uri">https://doi.org/10.48550/arXiv.2403.18602</a></p>
<p>Friedman, J., Hastie, T., &amp; Tibshirani, R. (2008). Sparse inverse
covariance estimation with the graphical lasso. <em>Biostatistics</em>,
9(3), 432–441. <a href="https://doi.org/10.1093/biostatistics/kxm045" class="uri">https://doi.org/10.1093/biostatistics/kxm045</a></p>
<p>Gross, S. M., &amp; Tibshirani, R. (2015). Collaborative regression.
<em>Biostatistics</em>, 16(2), 326–338. <a href="https://doi.org/10.1093/biostatistics/kxu047" class="uri">https://doi.org/10.1093/biostatistics/kxu047</a></p>
<p>Liu, H., Roeder, K., &amp; Wasserman, L. (2010). Stability Approach
to Regularization Selection (StARS) for High Dimensional Graphical
Models (arXiv:1006.3316). <em>arXiv</em> <a href="https://doi.org/10.48550/arXiv.1006.3316" class="uri">https://doi.org/10.48550/arXiv.1006.3316</a></p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
