<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Reproduction of manuscript’s results</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Reproduction of manuscript’s results</h1>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This vignette serves the purpose to reproduce the results shown in
the section “Real data example” of the manuscript describing
<em>collaborative graphical lasso</em> (<a href="#ref">Albanese, Kohlen
and Behrouzi, 2024</a>). The computations carried out by the following
code are heavy and could take a considerable amount of time. On our
machine (RAM 48 Gb, CPU Intel(R) Xeon(R) W-2235 CPU @ 3.80GHz) it took
almost 3 hours. Hence, we refer the user who wants to get started with
the package to the introductory vignette:
<code>vignette(&quot;coglasso&quot;)</code>. <em>Collaborative graphical
lasso</em> is an algorithm to reconstruct multi-omics networks based on
<em>graphical lasso</em> (<a href="#ref">Friedman, Hastie and
Tibshirani, 2008</a>) and <em>collaborative regression</em> (<a href="#ref">Gross and Tibshirani, 2015</a>). The coglasso package
implements and R interface to this algorithm. We will use it to
reproduce the results obtained in the original manuscript.</p>
<p>Let us first attach coglasso.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(coglasso)</span></code></pre></div>
<p>We will use the multi-omics data set provided by the package, in its
largest version: <code>multi_omics_sd</code>. Let’s inspect its
description in the manual.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">help</span>(multi_omics_sd)</span></code></pre></div>
<p>We notice that this data set has 162 transcripts and 76 metabolites.
For further information on the data set sources and generation, we refer
the user to the script <a href="https://github.com/DrQuestion/coglasso/blob/main/data-raw/multi_omics_sd.R">multi_omics_sd.R</a>
on the GitHub repository of the package.</p>
</div>
<div id="reproducing-the-results" class="section level2">
<h2>Reproducing the results</h2>
<p>For the multi-omics network reconstruction carried out in the
manuscript we explored 30 automatically generated <span class="math inline">\(λ_w\)</span> and <span class="math inline">\(λ_b\)</span> values and 9 given <span class="math inline">\(c\)</span> values.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>nlambda_w <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>nlambda_b <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>cs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.05</span>, <span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="dv">1</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">100</span>)</span></code></pre></div>
<p>We can now build the networks, one per combination of the chosen
hyperparameters, with <code>coglasso()</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>cg <span class="ot">&lt;-</span> <span class="fu">coglasso</span>(multi_omics_sd,</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  <span class="at">p =</span> <span class="dv">162</span>,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="at">nlambda_w =</span> nlambda_w,</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="at">nlambda_b =</span> nlambda_b,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  <span class="at">c =</span> cs</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>)</span></code></pre></div>
<p>We proceed now to select the most stable, yet sparse network, using
<em>eXtended StARS</em> (<em>XStARS</em>, <a href="#ref">Albanese,
Kohlen and Behrouzi, 2024</a>), an adaptation to coglasso of
<em>StARS</em> (<a href="#ref">Liu, Roeder and Wasserman, 2010</a>). The
function implementing <em>XStARS</em> is <code>select_coglasso()</code>,
when setting the argument <code>method</code> to “xstars” (which is the
default behaviour).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>sel_cg <span class="ot">&lt;-</span> <span class="fu">select_coglasso</span>(cg, <span class="at">method =</span> <span class="st">&quot;xstars&quot;</span>)</span></code></pre></div>
<p>We display now the selected network using the package
<code>igraph</code>, reproducing the one displayed in Figure 3 of the
manuscript. The orientation of the network may vary.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">plot</span>(sel_cg)</span></code></pre></div>
<p>We now plot the subnetwork of the gene <em>Cirbp</em> and of its
neighborhood. This reproduces the network displayed in Figure 4A.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>igraph<span class="sc">::</span><span class="fu">V</span>(sel_graph)<span class="sc">$</span>label<span class="ot">&lt;-</span><span class="fu">colnames</span>(sel_cg<span class="sc">$</span>data)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>cirbp_index <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colnames</span>(sel_cg<span class="sc">$</span>data) <span class="sc">==</span> <span class="st">&quot;Cirbp&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>subnetwork <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">subgraph</span>(</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  sel_graph,</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  <span class="fu">c</span>(cirbp_index, igraph<span class="sc">::</span><span class="fu">neighbors</span>(sel_graph, cirbp_index))</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>)</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="fu">plot</span>(subnetwork)</span></code></pre></div>
<p>A useful way to extract information from a network is community
discovery. This technique allows to simplify a network by breaking it in
functional units where nodes share several connections, enough to be
recognized as separate communities. In the manuscript we used the greedy
algorithm described in <a href="#ref">Clauset, Newman and Moore,
2004</a>, part of the <code>igraph</code> toolkit. We focused and
inspected the second largest community, shown in Figure 4B. Here is the
code to reproduce it.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>communities <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">cluster_fast_greedy</span>(sel_graph)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>community2 <span class="ot">&lt;-</span> communities[[<span class="dv">2</span>]]</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>community2_graph<span class="ot">&lt;-</span>igraph<span class="sc">::</span><span class="fu">subgraph</span>(sel_graph, community2)</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co"># Focusing on nodes of interest</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>fosjun_erg_AA <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Fos&quot;</span>, <span class="st">&quot;Fosb&quot;</span>, <span class="st">&quot;Junb&quot;</span>, <span class="st">&quot;Fosl2&quot;</span>, <span class="st">&quot;Egr1&quot;</span>, <span class="st">&quot;Egr2&quot;</span>, <span class="st">&quot;Egr3&quot;</span>, <span class="st">&quot;Ala&quot;</span>, <span class="st">&quot;Arg&quot;</span>, <span class="st">&quot;Asn&quot;</span>,<span class="st">&quot;His&quot;</span>,<span class="st">&quot;Ile&quot;</span>,<span class="st">&quot;Leu&quot;</span>,<span class="st">&quot;Lys&quot;</span>,<span class="st">&quot;Met&quot;</span>,<span class="st">&quot;Orn&quot;</span>,<span class="st">&quot;Phe&quot;</span>,<span class="st">&quot;Ser&quot;</span>,<span class="st">&quot;Thr&quot;</span>,<span class="st">&quot;Tyr&quot;</span>,<span class="st">&quot;Val&quot;</span>)</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>igraph<span class="sc">::</span><span class="fu">V</span>(community2_igraph)[<span class="sc">!</span>(label <span class="sc">%in%</span> fosjun_erg_AA)]<span class="sc">$</span>color<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;#3bd8ff&quot;</span>, <span class="dv">29</span>), <span class="fu">rep</span>(<span class="st">&quot;#ffadad&quot;</span>, <span class="dv">5</span>))</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>igraph<span class="sc">::</span><span class="fu">V</span>(community2_igraph)[<span class="sc">!</span>(label <span class="sc">%in%</span> fosjun_erg_AA)]<span class="sc">$</span>frame.color<span class="ot">&lt;-</span><span class="cn">NA</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>igraph<span class="sc">::</span><span class="fu">V</span>(community2_igraph)[<span class="sc">!</span>(label <span class="sc">%in%</span> fosjun_erg_AA)]<span class="sc">$</span>label<span class="ot">&lt;-</span><span class="cn">NA</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>lo_community2 <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">layout_with_fr</span>(community2_graph)</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="fu">plot</span>(community2_graph, <span class="at">layout=</span>lo_community2)</span></code></pre></div>
</div>
<div id="ref" class="section level2">
<h2>References</h2>
<p>Albanese, A., Kohlen, W., &amp; Behrouzi, P. (2024). Collaborative
graphical lasso (arXiv:2403.18602). <em>arXiv</em> <a href="https://doi.org/10.48550/arXiv.2403.18602" class="uri">https://doi.org/10.48550/arXiv.2403.18602</a></p>
<p>Friedman, J., Hastie, T., &amp; Tibshirani, R. (2008). Sparse inverse
covariance estimation with the graphical lasso. <em>Biostatistics</em>,
9(3), 432–441. <a href="https://doi.org/10.1093/biostatistics/kxm045" class="uri">https://doi.org/10.1093/biostatistics/kxm045</a></p>
<p>Gross, S. M., &amp; Tibshirani, R. (2015). Collaborative regression.
<em>Biostatistics</em>, 16(2), 326–338. <a href="https://doi.org/10.1093/biostatistics/kxu047" class="uri">https://doi.org/10.1093/biostatistics/kxu047</a></p>
<p>Liu, H., Roeder, K., &amp; Wasserman, L. (2010). Stability Approach
to Regularization Selection (StARS) for High Dimensional Graphical
Models (arXiv:1006.3316). <em>arXiv.</em> <a href="https://doi.org/10.48550/arXiv.1006.3316" class="uri">https://doi.org/10.48550/arXiv.1006.3316</a></p>
<p>Clauset, A., Newman, M. E. J., &amp; Moore, C. (2004). Finding
community structure in very large networks. <em>Physical Review E</em>,
70(6), 066111. <a href="https://doi.org/10.1103/PhysRevE.70.066111" class="uri">https://doi.org/10.1103/PhysRevE.70.066111</a></p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
